### ISSUE TITLE
cut_segments can only deal with ints for start and stop

### ISSUE THREAD

**Issue by cle1109** [scot-dev/scot#87]
We should make `cut_segments` more robust such that the function also works if floats are used for `start` and `stop` indices. The question is how do we handle floats? Should we just cast them to ints (thereby possibly truncating the numbers)? Or should we assert that a float is really just an int in disguise?

**Comment by mbillingr** [comment]
How do other libraries handle this?

**Comment by cle1109** [comment]
I just saw that we changed the code in the `transpose` branch to use `np.arange` instead of `range`. Maybe this solves the problem with data types, since `np.arange` also works with floats? However, I think the new code also fails if `start` and `stop` are actual floats that cannot be interpreted as ints...

**Comment by mbillingr** [comment]
`np.arange` does not really solve the issue. It's better than using range for other reasons, but we still need integer indices.

Actually, I think the best option would be to raise an error if start and stop cannot be safely interpreted as integers.
If users pass floats this could indicate that they misunderstood the meaning of the parameters and passed time instead of samples. Silently truncating the floats would give unexpected results and make the problem harder to debug, while a error message forces users to question their code.

**Comment by cle1109** [comment]
I agree that we should not silently truncate floats. However, it could be quite common that a user is well aware of the meaning of the parameters and still wants to pass floats. At least that's what I did when I wanted to pass in half a second; I computed that from 0.5 * fs, which is always a float (even though the result would actually be an integer number such as when fs = 256).

Would a check if the modulus by one resulted in zero be reasonable here?
